test/unit/route-regex.test.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-env jest */
+import { getRouteRegex } from 'next-server/dist/lib/router/utils/route-regex'
+
+describe('getRouteRegex', () => {
+  it('static route with param', async () => {
+    const { re, groups } = getRouteRegex('/api/test')
+    const matches = '/api/test?param=val'.match(re)
+
+    expect(matches).toHaveLength(1)
+    expect(matches[0]).toEqual('/api/test?param=val')
+    expect(groups).toEqual({})
+  })
+
+  it('dynamic route', async () => {
+    const { re, groups } = getRouteRegex('/api/[test]')
+    const matches = '/api/abc'.match(re)
+
+    expect(matches).toHaveLength(2)
+    expect(matches[0]).toEqual('/api/abc')
+    expect(matches[1]).toEqual('abc')
+    expect(groups).toEqual({ test: 1 })
```

Comment:
This must be failed. there is no { test: 1 }

Reply:
This is correct, `groups` is a map of the path param names and their positions, not the actual value. So this test and the other dynamic test have the same assertion for groups.

Reply:
Ok then.